src/services/PostEventProcessor.ts

```diff
@@ -41,164 +39,157 @@ export class PostEventProcessor {
+	async processRequest(eventBody: string): Promise<any> {
 		try {
 			const eventDetails: ReturnSQSEvent = JSON.parse(eventBody);
-			const eventName = eventDetails.event_name;
+			const { event_id, event_name, timestamp, user } = eventDetails;
 
-			if (!eventDetails.event_id) {
-				this.logger.error({ message: "Missing event_id in the incoming SQS event" }, { messageCode: MessageCodes.MISSING_MANDATORY_FIELDS });
-				throw new AppError(HttpCodesEnum.SERVER_ERROR, "Missing info in sqs event");
-			}
-			this.logger.appendKeys({ event_id: eventDetails.event_id });
+			const requiredFields = ['event_id', 'event_name', 'timestamp', 'user'];
 
-			this.logger.info({ message: "Received SQS event with eventName ", eventName });
-			if (!this.checkIfValidString([eventName]) || !eventDetails.timestamp) {
+			const missingFields = requiredFields.filter(field => !(field in eventDetails));
 
-				this.logger.error({ message: "Missing or invalid value for any or all of event name, timestamp in the incoming SQS event" }, { messageCode: MessageCodes.MISSING_MANDATORY_FIELDS });
-				throw new AppError(HttpCodesEnum.SERVER_ERROR, "Missing info in sqs event");
+			if (missingFields.length > 0) {
+				const errorMessage = "Missing fields in the incoming SQS event";
+				this.logger.error({ message: errorMessage }, { missingFields: missingFields.join(', ')}, { messageCode: MessageCodes.MISSING_MANDATORY_FIELDS });
+				throw new AppError(HttpCodesEnum.SERVER_ERROR, errorMessage);
 			}
-
-			if (!eventDetails.user) {
-				this.logger.error({ message: "Missing user details in the incoming SQS event" }, { messageCode: MessageCodes.MISSING_MANDATORY_FIELDS } );
-				throw new AppError(HttpCodesEnum.SERVER_ERROR, "Missing info in sqs event");
+	
+			const { user_id, govuk_signin_journey_id } = user;
+			if (!user_id) {
+					const errorMessage = "Missing or invalid value for userDetails.user_id in event payload";
+					this.logger.error({ message: errorMessage }, { messageCode: MessageCodes.MISSING_MANDATORY_FIELDS });
+					throw new AppError(HttpCodesEnum.SERVER_ERROR, errorMessage);
 			}
-			const userDetails = eventDetails.user;
-			this.logger.appendKeys({ govuk_signin_journey_id: userDetails.govuk_signin_journey_id });
-
-			if (!this.checkIfValidString([userDetails.user_id])) {
-				this.logger.error({ message: "Missing or invalid value for userDetails.user_id in event payload" }, { messageCode: MessageCodes.MISSING_MANDATORY_FIELDS });
-				throw new AppError(HttpCodesEnum.SERVER_ERROR, "Missing info in sqs event");
-			}
-
-			const userId = userDetails.user_id;
-			// Do not process the event if the event is already processed or flagged for deletion
-			const isFlaggedForDeletionOrEventAlreadyProcessed = await this.iprService.isFlaggedForDeletionOrEventAlreadyProcessed(userId, eventName);
+	
+			this.logger.appendKeys({ event_id, govuk_signin_journey_id });
+			this.logger.info({ message: `Received SQS event with eventName ${event_name}` });
```

Comment:
Reckon we could move this higher? Otherwise we are logging errors about the format of the event before we've even said that we have received it?

Reply:
Moved